### PR TITLE
refactor: Fix lint issues from latest flutter release

### PR DESCRIPTION
**Pull request by Salakar**
# Description

Updates to use the new color method `withValues`.

## Checklist

- [x] I have followed the [Contributor Guide] when preparing my PR.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples` or `docs`.

## Breaking Change?

- [ ] Yes, this PR is a breaking change.
- [ ] No, this PR is not a breaking change.

## Related Issues

[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Conventional Commit]: https://conventionalcommits.org
[CHANGELOG]: https://github.com/flame-engine/flame/blob/main/CHANGELOG.md

### CODE REVIEW

File: doc/flame/examples/lib/drag_events.dart [hunk 1]
```diff
@@ -127,7 +127,7 @@ class Trail extends Component {
       final path = _paths[i];
       final opacity = _opacities[i];
       if (opacity > 0) {
-        _linePaint.color = _color.withOpacity(opacity);
+        _linePaint.color = _color.withValues(alpha: opacity);
```
Comment (erickzanardo) [inline doc/flame/examples/lib/drag_events.dart:130]: I might be wrong, but iirc, alpha and opacity are not 1:1 changeable are they?

Comment (erickzanardo) [reply doc/flame/examples/lib/drag_events.dart:130]: Oh, nvm: https://docs.flutter.dev/release/breaking-changes/wide-gamut-framework#withopacity-migration

File: packages/flame/test/effects/color_effect_test.dart [hunk 2]
```diff
@@ -18,7 +18,11 @@ void main() {
       game.update(0);
       expect(
         component.paint.colorFilter.toString(),
-        equals('ColorFilter.mode(Color(0x00f44336), BlendMode.srcATop)'),
+        equals(
```
Comment (Salakar) [inline packages/flame/test/effects/color_effect_test.dart:21]: This is super sus - not sure this is right approach ofc but values seem rounded

Comment (spydon) [reply packages/flame/test/effects/color_effect_test.dart:21]: Interesting that we compare it with the string, maybe that's the easiest way to check it.
Let's keep it like this for now.

**Review by spydon — APPROVED** [review]
Lgtm!

File: packages/flame_3d/lib/src/extensions/color.dart [hunk 3]
```diff
@@ -4,9 +4,9 @@ import 'dart:ui';
...
   /// Returns a Float32List that represents the color as a vector.
   Float32List get storage => Float32List.fromList([
-        opacity,
-        red.toDouble() / 255,
-        green.toDouble() / 255,
-        blue.toDouble() / 255,
+        a,
+        r / 255,
+        g / 255,
+        b / 255,
```
Comment (wolfenrain) [inline packages/flame_3d/lib/src/extensions/color.dart:10]:
```suggestion
        a,
        r,
        g,
        b,
```

According to the docs these are already in normalized space

**Review by wolfenrain — APPROVED** [review]

File: packages/flame_3d/lib/src/extensions/color.dart (hunk 3 above)
Comment (Salakar) [reply packages/flame_3d/lib/src/extensions/color.dart]: derp, good catch

File: packages/flame_3d/lib/src/graphics/graphics_device.dart [hunk 4]
```diff
@@ -177,7 +177,15 @@ class GraphicsDevice {
       );
 
       _renderTarget = gpu.RenderTarget.singleColor(
-        gpu.ColorAttachment(texture: colorTexture!, clearValue: clearValue),
+        gpu.ColorAttachment(
```
Comment (Salakar) [inline packages/flame_3d/lib/src/graphics/graphics_device.dart:180]: assuming the vector expected was in the order RGBA

Comment (wolfenrain) [reply packages/flame_3d/lib/src/graphics/graphics_device.dart:180]: Yes it is RGBA (probably)

**Review by wolfenrain — APPROVED** [review]

**Pull request merged**